apisix/plugin.lua

```diff
@@ -875,14 +875,20 @@ function _M.run_plugin(phase, plugins, api_ctx)
         return api_ctx
     end
 
+    local ori_phase = phase
     if phase ~= "log"
         and phase ~= "header_filter"
         and phase ~= "body_filter"
         and phase ~= "delayed_body_filter"
     then
         for i = 1, #plugins, 2 do
-            if phase == "rewrite_in_consumer" and plugins[i + 1]._from_consumer
-                    and plugins[i].type ~= "auth"then
+            if ori_phase == "rewrite_in_consumer" and not plugins[i + 1]._from_consumer
+                    and plugins[i].type == "auth" then
+                plugins[i + 1]._skip_rewrite_in_consumer = true
+            end
+
+            if ori_phase == "rewrite_in_consumer" and plugins[i + 1]._from_consumer
```

Comment:
Will it be more readable if we use a local `xxx_phase` variable and keep the `phase` passed as the argument unchanged? The reasons are:
1. better not to rename the variable which is passed as the argument
2. the new `xxx_phase`'s scope will not pass through a step in  `for i = 1, #plugins, 2 do` loop. It won't affect the next plugin.

Reply:
ok, minor refactoring, now the local `xxx_phase` variable is not needed and the `phase` is not modified.